### ISSUE TITLE
TravisCI performance

### ISSUE THREAD

**Issue by damianavila** [bokeh/bokeh#1084]
The idea here is to propose some mechanism to improve the TravisCI situation and the row of builds we have every day waiting to be tested.
I have proposed to test only the PRs and deactivate the build of pushes/merge because we are testing things twice. @mattpap has raised some comments about problem with image diff if we deactivate the pushes/merge, @mattpap can you summarize here the problem?

As a plus, we can maybe start discussing the big pile of examples we are testing and question ourselves... is it really necessary to test all the examples? would not be more efficient just to test some paradigmatic ones and each build, and make a general "local" before hot spots like releases?

Finally, ideas to improve the situation in another way, for instance, parallelization of testing? or other?

Everyone is invited to the discussion :wink:

**Comment by birdsarah** [comment]
Discussion is now more than 3 months old. Can re-open if renewed interest.
